src/Tasks/DownloadFile.cs

```diff
@@ -148,9 +148,13 @@ private async Task DownloadAsync(Uri uri, CancellationToken cancellationToken)
                     }
                     catch (HttpRequestException e)
                     {
-                        // HttpRequestException does not have the status code so its wrapped and thrown here so that later on we can determine
-                        // if a retry is possible based on the status code
+#if RUNTIME_TYPE_NETCORE
```

Comment:
Not sure what the benefit is going with NET. RUNTIME_TYPE_NETCORE isn't great either but the use-case is clear. Is `NET` safer to use?

Reply:
I don't know which preprocessor directives include each other, but this should be for net6, right? That sounds like NET to me, even if RUNTIME_TYPE_NETCORE includes NET.

Reply:
Turns out this is actually better off as `NET6_0_OR_GREATER` because the new API started in net5.0. It would cause issues in the ns2.0 build of Microsoft.Build.Tasks.

Come to think of it, I'm not sure how M.B.Tasks passed during the ns2.0 build. Should be fixed now though.